api/task.go

```diff
@@ -122,11 +122,33 @@ func (h *taskHandler) updateTask(w http.ResponseWriter, r *http.Request) {
 
 	ctx := context.Background()
 
-	plan, err := d.UpdateTask(ctx, patch)
-	if err != nil {
+	var storedErr error
+	if runOp == driver.RunOptionNow {
+		task := d.Task()
+		ev, err := event.NewEvent(taskName, &event.Config{
+			Providers: task.ProviderNames(),
+			Services:  task.ServiceNames(),
+			Source:    task.Source(),
+		})
+		if err != nil {
+			log.Printf("[ERROR] (api.task) error creating event for task %q: %s",
+				taskName, err)
+		}
+		defer func() {
+			ev.End(storedErr)
+			log.Printf("[TRACE] (api.task) adding event %s", ev.GoString())
+			if err := h.store.Add(*ev); err != nil {
```

Comment:
Minor: The `defer func` seems to be initialized even if there is an error, where `ev` may be nil, then it would panic here. It's quite unlikely `NewEvent` will ever error, but can you update this where panics can't be a part of a possible path? Thanks!

Reply:
Good point! Thanks for raising this! Some above code checks for the edge-case that would cause `NewEvent` to return error i.e. when taskName is empty. But I'm realizing that this dependency isn't clear at all and that code may evolve in a way that opens up for more panic. I decided to add a return when there's an error creating an event. Thank you!